src/model/operation/mergeoperation.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module engine/model/operation/mergeoperation
+ */
+
+import Operation from './operation';
+import SplitOperation from './splitoperation';
+import Position from '../position';
+import Range from '../range';
+import { _remove, _move } from './utils';
+
+import CKEditorError from '@ckeditor/ckeditor5-utils/src/ckeditorerror';
+
+/**
+ * Operation to merge two {@link module:engine/model/element~Element elements}. The merged elements are a parent of given
+ * {@link module:engine/model/position~Position position} and the next element.
+ *
+ * Technically, the content of the next element is moved at given {@link module:engine/model/position~Position position}
+ * and the element is removed.
+ *
+ * @extends module:engine/model/operation/operation~Operation
+ */
+export default class MergeOperation extends Operation {
+	/**
+	 * Creates a merge operation.
+	 *
+	 * @param {module:engine/model/position~Position} sourcePosition Position inside the merged element. All nodes from that
+	 * element after that position will be moved to {@link ~#targetPosition}.
+	 * @param {module:engine/model/position~Position} targetPosition Position which the nodes from the merged elements will be moved to.
```

Comment:
Alternatively, this could be positions before merged element and after to-merge element.